### PR TITLE
Updates to compiler docs

### PR DESCRIPTION
**Pull request by nbhuiyan**
Problem determination document contains info about verbose log and compilation trace logs. Compiler options now have a dedicated document. A lot of the contents in the PR have been migrated from the internal wiki with some changes.

Anyone looking at this PR: please feel free to make suggestions for additions/deletions, point out incorrect info, etc. The goal of these changes is that the community outside of IBM have easy access to information when working with the OMR Compiler.

Signed-off-by: Nazim Uddin Bhuiyan <nazimudd@ualberta.ca>

### CODE REVIEW

**Comment by nbhuiyan** [comment]
@xliang6

File: doc/compiler/ProblemDetermination.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ contains a potentially faulty optimization. If the program still fails at
 `noOpt`, then the problem is most likely in the code generation phase (versus
 the optimization phase) of the compiler.
```
Comment (xliang6) [inline doc/compiler/ProblemDetermination.md]: I think it's worth mentioning early on in the document that you can change opt levels by specifying `-Xjit:optlevel=noOpt` command line or setting `TR_Options='optLevel=noOpt'` and then refer to the doc/compiler/CompilerOptions.md file for more details.

**Comment by nbhuiyan** [comment]
> I'm not particularly keen on exhaustively listing each compiler option, and especially if you admit that list is not exhaustive and not guaranteed to be correct. I would just refer the interested reader to the tables in Options.cpp and FEBase.cpp

@0xdaryl, between listing all of these options as they are here, and only listing just a few commonly used ones as examples, what do you think is a good balance (given that we will have a link to `OMROptions.hpp` and `FEBase.cpp` in the doc) ?

@xliang6 any thoughts on that?

**Comment by xliang6** [comment]
I would go with listing a few commonly used ones and also adding the ones that are not self-explanatory, then link to `OMROptions.hpp` for the complete list. Thanks.

**Comment by nbhuiyan** [comment]
@xliang6 @0xdaryl Ready for another review. To preview the documents:
 1. [ProblemDetermination.md](https://github.com/nbhuiyan/omr/blob/48904e138d73f13b0e2b01011a0c00fa02fbfc0e/doc/compiler/ProblemDetermination.md)
2. [CompilerOptions.md](https://github.com/nbhuiyan/omr/blob/48904e138d73f13b0e2b01011a0c00fa02fbfc0e/doc/compiler/CompilerOptions.md)

File: doc/compiler/CompilerOptions.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+
+# Option Sets
+
+It is possible to apply a set of compiler options ("option sets") only to a
+method, or group of methods. Furthermore, it is possible to apply different
+option sets to different (groups of) methods. An option set can be defined on
+the command line by enclosing the desired set of parameters in parentheses,
+and preceding it with a method filter expression, e.g.
```
Comment (0xdaryl) [inline doc/compiler/CompilerOptions.md]: ...and preceding it with a method filter expression in braces.  For example, to apply the option set `count=0,optlevel=hot` only to the method that matches the signature `HelloWorld.main([Ljava/lang/String;)V` add the following to the `TR_Options` environment variable or `-Xjit` command-line:

File: doc/compiler/CompilerOptions.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+```sh
+{HelloWorld.main([Ljava/lang/String;)V}(count=0,optlevel=hot)
+```
+
+The option example below makes the compiler compile all methods at the warm
+optimization level, and trace the compilation of all methods that start
```
Comment (0xdaryl) [inline doc/compiler/CompilerOptions.md]: * ... and traces ...

* ... all method signatures ...

File: doc/compiler/CompilerOptions.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+```sh
+optlevel=warm,count=0,{Frobnicate*}(log=frob.log,traceFull),{Brobnicate*}(log=brob.log,traceFull)
+```
+
+Option sets can also be defined in a limit file, if one is used, by adding a
+non-zero integer immediately after the initial plus sign on the line with the
+selected method(s).
+
+```sh
```
Comment (0xdaryl) [inline doc/compiler/CompilerOptions.md]: Maybe move this down into your example below.

File: doc/compiler/ProblemDetermination.md [hunk 5]
```diff
@@ -229,286 +244,282 @@ when it ends. If the compiler fails on a particular method (that is, it starts
...
+    |   |   |     |     -------------- Global Index
+    |   |   |     |     |     ------------ Side Table Index
+    |   |   |     |     |     |   ------------- Use/def Index
+    |   |   |     |     |     |   |  ------------- Number of Children
+    |   |   |     |     |     |   |  |  ------------- Size
+    |   |   |     |     |     |   |  |  |      ------------- Data Type
+    |   |   |     |     |     |   |  |  |      |           ------------- Node Address
+    |   |   |     |     |     |   |  |  |      |           |      ------------- Instruction
+    |   |   |     |     |     |   |  |  |      |           |      |
+    V   V   V     V     V     V   V  V  V      V           V      V
```
Comment (fjeremic) [inline doc/compiler/ProblemDetermination.md]: This section is outdated. The traces do not look like this anymore. Can we update to how they look today?

**Comment by nbhuiyan** [comment]
Ready for another review

**Pull request merged**